core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2103,6 +2103,13 @@ private[zio] trait ZIOFunctions extends Serializable {
       loop
     }
 
+  /**
+   * Applies the function `f` if the argument is non-empty and
+   * returns the results in a new `Option[B]`.
+   */
+  final def foreachOption[R, E, A, B](in: Option[A])(f: A => ZIO[R, E, B]): ZIO[R, E, Option[B]] =
```

Comment:
I guess this also needs the discarding variants, `foreach/traverseOption_`

Reply:
Actually, there's no difference between a `foreachOption_` and `foreach`, I guess a discarding variant is redundant in anything but naming consistency.